File: 409-LongestPalindrome/409-LongestPalindrome.cpp
// Last updated: 7/26/2026, 10:58:37 PM
class Solution {
public:
    int longestPalindrome(string s) {
        unordered_map<char, int> map;

        for (int i = 0; i < s.length(); i++) {
            char ch = s[i];
            map[ch]++;
        }

        bool flag = false;
        int c = 0;

        for (auto &p : map) {
            char ch = p.first;
            int freq = p.second;

            if (freq > 1) {
                if (freq % 2 == 0) {
                    c += freq;
                } 
                else if (!flag) {
                    c += freq;
                    flag = true;
                } 
                else {
                    c += (freq - 1);
                }
            } 
            else {
                if (!flag && freq == 1) {
                    flag = true;
                    c++;
                }
            }
        }

        return c;
    }
};

File: 409-LongestPalindrome/409-LongestPalindrome.java
// Last updated: 7/26/2026, 10:58:41 PM
class Solution {
    public int longestPalindrome(String s) {
        HashMap<Character,Integer> map=new HashMap<>();
        for(int i=0;i<s.length();i++){
            char ch=s.charAt(i);
            map.put(ch,map.getOrDefault(ch,0)+1);
        }
        boolean flag=false;
        int c=0;
        for(char ch:map.keySet()){
            if(map.get(ch)>1){
                if(map.get(ch)%2==0){
                    c+=map.get(ch);
                }else if(!flag){
                    c=c+(map.get(ch));
                    flag=true;
                }else{
                    c=c+(map.get(ch)-1);
                }
            }else{
                if(!flag && map.get(ch)==1){
                    flag=true;
                     c++;
                }
            }
        }
        return c;
    }
}